src/rust/engine/fs/store/src/lib.rs

```diff
@@ -1100,7 +1100,9 @@ impl Store {
   ) -> BoxFuture<'static, Result<Vec<DigestEntry>, String>> {
     self
       .walk(digest, move |_, path_so_far, _, directory| {
-        if directory.files.is_empty() {
+        if digest == EMPTY_DIGEST {
+          future::ready(Ok(vec![])).boxed()
+        } else if directory.files.is_empty() {
```

Comment:
This can be outside of the `walk` call at the very top of this function. The `digest` won't change.

Reply:
Y

Reply:
Y